File: LeetCode_Algorithm/66/66.py
from typing import List

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        # sol1 O(n), O(n) 39~44ms 70~57%
        for i in range(len(digits)-1 ,-1, -1):
            digits[i] = digits[i]+1
            
            if digits[i] == 10:
                digits[i] = 0; 
                if i == 0:
                    return [1]+digits
            else:
                break
        return digits

print(Solution().plusOne([9,9,9]))
            
        
        

File: LeetCode_Algorithm/66/66.js
var plusOne = function(digits) {
// sol1 O(n), O(n) 61~67ms 86~74% 
  for (let i=digits.length-1; i > -1 ; i--) {
    digits[i] += 1
    if (digits[i] == 10) {
      digits[i] = 0
      if (i == 0) { 
        digits.unshift(1)
        return digits
      }
    }
    else  {
      break
    }
  }
  return digits
};

console.log(plusOne([9,9,9]))